classes/local/client/s3_client.php

```diff
@@ -164,6 +193,41 @@ public function test_connection() {
         return $connection;
     }
 
+    /**
+     * Tests permissions to write to S3 bucket.
+     *
+     * @return object
+     */
+    public function test_permissions() {
+        $permissions = new \stdClass();
+
+        $permissions->success = true;
+        $permissions->messages = [];
+
+        if ($this->is_functional()) {
+            $permissions->success = false;
+            $permissions->messages = [];
```

Comment:
Do we need to add a message here that it's not functional?

Reply:
This should be covered by test connection. So we don't need messages here.